### ISSUE TITLE
Output haplotypes

### ISSUE THREAD

**Issue by aziouez25** [rwdavies/STITCH#10]
Hello Robert,

Is there a way to output the estimated recombination rate positions?

Thanks
a

**Comment by rwdavies** [comment]
Hey, 

Recombination rate is estimated between the positions you supply STITCH. This output is not "supported" currently but is available in a grab-bag of internal parameters and results saved in RData/EM.all.<regionName>.RData. 

You want "sigmaCurrent" and can line it up using "pos" as follows

```
> head(pos)
    CHR      POS REF ALT
1 chr19 10000105   T   G
2 chr19 10000345   T   C
3 chr19 10000612   C   T
4 chr19 10000968   G   T
5 chr19 10001946   A   G
6 chr19 10002175   T   C
> head(sigmaCurrent)
[1] 0.9998975 0.9998979 0.9995245 0.9990623 0.9997978 0.9999408
```
so here, 0.9998975 is between 10000105 and 10000345. These values are stored internally in sigmaCurrent as 
exp(-nGen * rate{cM/Mb} / 100 / 1000000 * distance{between SNPs in bp})
therefore if you want an estimate of the rate in cM/Mb between SNPs, then do something like 

 log(sigmaCurrent) / -nGen * 100 * 1000000 / diff(pos[, 2])

Note that values are capped internally during updating by minRate and maxRate which have default values but you can choose different values for those

Best,
Robbie

**Comment by aziouez25** [comment]
Great! Thanks for your quick reply.
My problem is that I want to tag a multi-allelic indel (5 alleles) in the UKTwins with haplotypes instead of variants because 2 alleles are rare. I took the 2000 closest SNVs to that indel regardless of their MAF and ran STITCH generating 30 haplotypes (--K=30 and default parameters).
I removed the SNVs involving with the same frequency in all haplotypes or frequency <0.05 or >0.95 in any haplotypes (last eHapsCurrent_t) keeping 30 haplotypes of 557 SNVs and ran a decision tree for tagging. Obviously it did not work so I wanted to play with recombination sites and other heuristics like --K, MAF, number of SNVs..

I know STITCH is not dedicated to this kind of tasks, but any advise would be more than welcome!

Thanks again
a

**Comment by aziouez25** [comment]
Hi Roby,
Many thanks for your reply. 
May be my message was not very clear. Sorry for that. 
The coverage is pretty high: >30X. The detection of the alleles was done by Pacbio on different samples. Calling the alleles on UKtwins was not the problem. I wrote a fasta file with 5 fake chromosomes; one fake chromosome per allele. I run bwa and called with GATK HC. HWE was respected then, but I will rerun Platypus to check again.
Concerning STITCH, I want to use it for Haplotype construction only. I am not interested in the imputation.  My idea was to tag the 5 alleles with haplotypes instead of SNPs for future use on SNP arrays. So, I first extracted the 2,000 closest SNPs to my indel but that was too much. I then extracted the 50 closest SNPs and ran STITCH with --K=50. The problem is that I have many "duplicated" haplotypes and also haplotypes with "unknown" genotypes (eHapsCurrent_t[i,]>0.1 & e_hapsCurrent_t[i,] <0.9). 

My question is how to exclude these "unknown" genotypes from the last e_hapsCurrent_t?

Thanks
A
